### PR TITLE
Add parsing of proto values

### PR DESCRIPTION
**Pull request by joshuaghezzi**
- Reading values into data dictionary from proto.
- Currently just enums for now.
- Updating go version to 1.22.
- Removing temp fix.

### CODE REVIEW

File: action.sh [hunk 1]
```diff
@@ -35,6 +35,9 @@ fi
 if [[ -z "$REQUIRED_ROLES" ]]; then
     echo "WARNING: missing REQUIRED_ROLES, consider adding them to the action config"
 fi
+if [[ -z "$PROTO_PATH" ]]; then
+    echo "WARNING: missing PROTO_PATH, skipping reading proto files for values"
```
Comment (gouline) [inline action.sh]: We don't require proto, so it shouldn't be a warning, e.g. INFO.

File: README.md [hunk 2]
```diff
@@ -10,12 +10,13 @@ GitHub Action for generating and checking freshness of `data.json` data dictiona
...
     - [`rubenv-sql-migrate`](https://github.com/rubenv/sql-migrate)
 * `tool-path` - Path to the data tool files (e.g. migration).
+* `proto-path` - Path to services proto files.
 * `dictionary-app-id` - Dictionary GitHub App ID.
 * `dictionary-app-private-key` - Dictionary GitHub App private key.
-* `repo-token` - GitHub auth token for pull request comments (NOT for commits). Defaults to `${{ github.token }}`.
 * `dictionary-repo` - Dictionary repo name. Defaults to `data-dictionary`.
 * `dictionary-ref` - Dictionary branch reference. Defaults to `master`.
 * `dictionary-workflow` - Dictionary workflow name. Defaults to `build.yml`.
+* `repo-token` - GitHub auth token for pull request comments (NOT for commits). Defaults to `${{ github.token }}`.
```
Comment (gouline) [inline README.md:19]: Why move?

File: action/proto/proto.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+        Args:
+            workspace (Path): The path of the workspace.
+            proto_path_str (str, optional): The path to the proto files.
+        """
+        self.proto_code_dict = {}
+        if proto_path_str:
+            proto_path = workspace / proto_path_str
+            for proto_file_path in proto_path.glob("**/*.proto"):
+                with open(proto_file_path, "r", encoding="utf-8") as proto_file:
```
Comment (gouline) [inline action/proto/proto.py]: I think `"r"` is the default, so you can omit it

File: action/proto/proto.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    def __init__(self, workspace: Path, proto_path_str: Optional[str] = ""):
+        """Create a new ProtoReader instance.
+
+        Args:
+            workspace (Path): The path of the workspace.
+            proto_path_str (str, optional): The path to the proto files.
+        """
+        self.proto_code_dict = {}
+        if proto_path_str:
+            proto_path = workspace / proto_path_str
```
Comment (gouline) [inline action/proto/proto.py]: Redundant var, right? You're only using it in the loop

File: action/proto/proto.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+class ProtoReader:
+    def __init__(self, workspace: Path, proto_path_str: Optional[str] = ""):
+        """Create a new ProtoReader instance.
+
+        Args:
+            workspace (Path): The path of the workspace.
+            proto_path_str (str, optional): The path to the proto files.
+        """
+        self.proto_code_dict = {}
+        if proto_path_str:
```
Comment (gouline) [inline action/proto/proto.py]: What happens when it's empty? Does it even do anything then?

Also, Python convention is accepting a path as `proto_path: Union[str, Path]` and then `proto_path = Path(proto_path)` to ensure it's a `Path`.

File: action/proto/proto.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            proto_path = workspace / proto_path_str
+            for proto_file_path in proto_path.glob("**/*.proto"):
+                with open(proto_file_path, "r", encoding="utf-8") as proto_file:
+                    self.proto_code_dict[proto_file.name] = proto_file.read()
+
+    def read(self) -> dict[str, dict[str, str]]:
+        """Read in values from proto code.
+
+        returns dict(str, dict[str,str]): Parsed proto values.
```
Comment (gouline) [inline action/proto/proto.py]: Incorrectly formatted doc line

File: action/proto/proto.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+        proto_ast = proto_parser.Parser().parse(proto_code)
+        enums = {}
+        for enum_element in filter(
+            lambda x: isinstance(x, Enum), proto_ast.file_elements
+        ):
+            enums[enum_element.name] = {
+                enum_value.name: enum_value.number
+                for enum_value in enum_element.elements
+                if isinstance(enum_value, EnumValue)
```
Comment (gouline) [inline action/proto/proto.py]: Strange combination of loop and dictionary comprehension. If you have an outer loop, this should probably be a loop with a filter as well

File: requirements.txt [hunk 8]
```diff
@@ -1 +1,2 @@
 requests
+proto_schema_parser==1.3.3
```
Comment (gouline) [inline requirements.txt]:
```suggestion
proto_schema_parser~=1.3.3
```

File: README.md (hunk 2 above)
Comment (joshuaghezzi) [reply README.md:19]: Just cause it was in the middle of the `dictionary-` inputs.

File: action/proto/proto.py (hunk 5 above)
Comment (joshuaghezzi) [reply action/proto/proto.py]: Yeah, my thinking was when we don't pass any path then just do nothing

Comment (gouline) [reply action/proto/proto.py]: If it does nothing without that path, why instantiate this class at all?

**Review by gouline — APPROVED** [review]

File: action/proto/proto.py (hunk 5 above)
Comment (joshuaghezzi) [reply action/proto/proto.py]: Yeah, that makes sense. Changed it so it only instantiates it if a path was actually passed.

**Pull request merged**